Lib/idlelib/editor.py

```diff
@@ -1295,38 +1295,53 @@ def smart_indent_event(self, event):
             text.undo_block_stop()
 
     def newline_and_indent_event(self, event):
+        """Insert a newline and indentation after Enter keypress event.
+
+        Properly position the cursor on the new line based on information
+        from the current line.  This takes into account if the current line
+        is a shell prompt, is empty, has selected text, contains a block
+        opener, contains a block closer, is a continuation line, or
+        is inside a string.
+        """
         text = self.text
         first, last = self.get_selection_indices()
         text.undo_block_start()
+        # This is wrapped in a try so that the finally clause is always
```

Comment:
IMO, this comment is unnecessary.

Reply:
It would have been added because the finally clause is 100 lines away.  I am replacing the two lines with a more informative comment on the `try:` line.